spec/controllers/openid_connect/authorizations_controller_spec.rb

```diff
@@ -0,0 +1,172 @@
+require 'spec_helper'
+
+describe OpenidConnect::AuthorizationsController, type: :controller do
+  let!(:client) { OAuthApplication.create!(name: "Diaspora Test Client", redirect_uris: ["http://localhost:3000/"]) }
+  let!(:client_with_multiple_redirects) { OAuthApplication.create!(name: "Diaspora Test Client", redirect_uris: ["http://localhost:3000/","http://localhost/"]) }
+
+  before do
+    sign_in :user, alice
+    allow(@controller).to receive(:current_user).and_return(alice)
+    Scope.create!(name:"openid")
+  end
+
+  describe "#new" do
+    context "when valid parameters are passed" do
+      render_views
+      context "as GET request" do
+        it "should return a form page" do
+          get :new,
+              {
+                client_id: client.client_id,
+                redirect_uri: "http://localhost:3000/",
+                response_type: "id_token",
+                scope: "openid",
+                nonce: SecureRandom.hex(16),
+                state: SecureRandom.hex(16)
+              }
+          expect(response.body).to match("Diaspora Test Client")
+        end
+      end
+      context "as POST request" do
+        it "should return a form page" do
+          post :new,
+              {
+                client_id: client.client_id,
+                redirect_uri: "http://localhost:3000/",
+                response_type: "id_token",
+                scope: "openid",
+                nonce: SecureRandom.hex(16),
+                state: SecureRandom.hex(16)
+              }
+          expect(response.body).to match("Diaspora Test Client")
+        end
+      end
+    end
+    context "when client id is missing" do
+      it "should return an bad request error" do
+        post :new,
+             {
+               redirect_uri: "http://localhost:3000/",
+               response_type: "id_token",
+               scope: "openid",
+               nonce: SecureRandom.hex(16),
+               state: SecureRandom.hex(16)
+             }
+        expect(response.body).to match("bad_request")
+      end
+    end
+    context "when redirect uri is missing" do
+      context "when only one redirect URL is pre-registered" do
+        it "should return a form pager" do
+          # Note this intentionally behavior diverts from OIDC spec http://openid.net/specs/openid-connect-core-1_0.html#AuthRequest
+          # See https://github.com/nov/rack-oauth2/blob/master/lib/rack/oauth2/server/authorize.rb#L63
```

Comment:
I don't think the referenced line doesn't make it obvious why we need to violate the OIDC spec, care to elaborate? (in the comment)